packages/x-components/src/x-modules/search/components/__tests__/banner.spec.ts

```diff
@@ -61,6 +61,25 @@ describe('testing Banner component', () => {
 
     expect(wrapper.get(getDataTestSelector('banner')).text()).toEqual('Search UIs');
   });
+
+  // eslint-disable-next-line max-len
+  it('emits UserClickedABanner when the user clicks in the left, middle or right button on the component', () => {
+    const listener = jest.fn();
+    const banner = createBannerStub('banner');
+    const { wrapper } = renderBanner({ banner });
+    wrapper.vm.$x.on('UserClickedABanner').subscribe(listener);
+
+    wrapper.trigger('click');
```

Comment:
quick question, why we don't need an `await` before trigger here? We tend to use it and checking the [docs](https://v1.test-utils.vuejs.org/guides/dom-events.html) they're using it also since its returning a Promise 🤔   
Maybe this can lead to unexpected async errors

Reply:
The click event is emitted immediately. We don't need to wait for another re-render which is what the await. If it changes and is emitted asynchronously somehow I would like to be informed.

Reply:
Salu2